### PR TITLE
add test for etc customization

### PR DESCRIPTION
**Pull request by yih-redhat**

### CODE REVIEW

File: check-ostree.yaml [hunk 1]
```diff
@@ -1019,7 +1020,85 @@
       when:
         - (ansible_facts['distribution'] == 'RedHat' and ansible_facts ['distribution_version'] is version('9.2', '>=')) or
           ((ansible_facts['distribution'] == 'RedHat' and ansible_facts ['distribution_version'] is version('8.8', '>=')) and ansible_facts ['distribution_version'] is version('9.0', '!=') and ansible_facts ['distribution_version'] is version('9.1', '!=')) or
-          (ansible_facts['distribution'] == 'CentOS')
+          (ansible_facts['distribution'] == 'CentOS' and ansible_facts ['distribution_version'] is version('8', '!='))
```
Comment (henrywang) [inline check-ostree.yaml:1023]: Could you please remove this line of code? Thanks.

**Review by henrywang — CHANGES_REQUESTED** [review]

**Comment by henrywang** [comment]
@yih-redhat All Fedora tests are failed. Could you please take a look? Does this feature support Fedora? Thanks.

**Comment by yih-redhat** [comment]
> @yih-redhat All Fedora tests are failed. Could you please take a look? Does this feature support Fedora? Thanks.

Sure, I will take a look and fix it.

**Comment by yih-redhat** [comment]
> @yih-redhat All Fedora tests are failed. Could you please take a look? Does this feature support Fedora? Thanks.

oh, fedora does not have version 75, so this feature test will fail.

**Comment by henrywang** [comment]
@yih-redhat What's the status of this PR? Is osbuild-composer v75 available on Fedora? If yes, please re-run the fedora tests. If the feature is not supported by fedora, please remove this part of test on fedora.

**Comment by henrywang** [comment]
@yih-redhat could you please make a rebase and run again? Fedora and CS8 should work now.

File: check-ostree.yaml (hunk 1 above)
Comment (yih-redhat) [reply check-ostree.yaml]: Removed.

**Comment by henrywang** [comment]
@yih-redhat I re-run f38 cases. Could you please take a look at the failure? Thanks.

**Comment by yih-redhat** [comment]
> @yih-redhat I re-run f38 cases. Could you please take a look at the failure? Thanks.

Command "systemctl status custom.service" has different output on fedora 37 and 38, that why the failure happens.

On Fedora 38:
[admin@localhost ~]$ systemctl status custom.service
● custom.service - Custom service
     Loaded: loaded (/etc/systemd/system/custom.service; enabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
             /etc/systemd/system/custom.service.d
             └─override.conf
     Active: active (exited) since Wed 2023-03-22 13:24:37 UTC; 2h 3min ago
    Process: 695 ExecStart=/usr/bin/cat /etc/custom_file.txt (code=exited, status=0/SUCCESS)
   Main PID: 695 (code=exited, status=0/SUCCESS)
        CPU: 2ms

Mar 22 13:24:37 localhost.localdomain systemd[1]: Starting custom.service - Custom service...
Mar 22 13:24:37 localhost.localdomain cat[695]: image builder is the best
Mar 22 13:24:37 localhost.localdomain systemd[1]: Finished custom.service - Custom service.

On Fedora 37:
[admin@localhost ~]$ systemctl status custom.service 
● custom.service - Custom service
     Loaded: loaded (/etc/systemd/system/custom.service; enabled; preset: disabled)
    Drop-In: /etc/systemd/system/custom.service.d
             └─override.conf
     Active: active (exited) since Wed 2023-03-22 15:22:24 UTC; 43s ago
    Process: 724 ExecStart=/usr/bin/cat /etc/custom_file.txt (code=exited, status=0/SUCCESS)
   Main PID: 724 (code=exited, status=0/SUCCESS)
        CPU: 2ms

Mar 22 15:22:24 localhost.localdomain systemd[1]: Starting custom.service - Custom service...
Mar 22 15:22:24 localhost.localdomain cat[724]: image builder is the best
Mar 22 15:22:24 localhost.localdomain systemd[1]: Finished custom.service - Custom service.


The output on Fedora 38, in Drop-In section, has one more additional path than Fedora 37, that's why the DropInPath check fail. I removed this DropInPath check because other checks can make sure custom.service works as expected.

**Comment by yih-redhat** [comment]
@henrywang Can you please merge this pr?

**Comment by henrywang** [comment]
@yih-redhat I found you removed `Check that 'custom.service' was overridden by drop-in` task in playbook. The drop-in is one of checking point. RHEL, CS and Fedora 37 still work with this test. Could you please add a `when` condition checking and keep this case? You can have another task for Fedora 38 with two drop-ins. Thanks.

**Comment by yih-redhat** [comment]
@henrywang all test passed, can you please merge it?

**Review by henrywang — APPROVED** [review]

**Pull request merged**